clang/lib/Frontend/DependencyFile.cpp

```diff
@@ -312,19 +316,26 @@ static void PrintFilename(raw_ostream &OS, StringRef Filename,
 }
 
 void DependencyFileGenerator::outputDependencyFile(DiagnosticsEngine &Diags) {
-  if (SeenMissingHeader) {
-    llvm::sys::fs::remove(OutputFile);
+  Expected<llvm::vfs::OutputFile> O = OutputBackend->createFile(
+      OutputFile,
+      llvm::vfs::OutputConfig().setTextWithCRLF().setNoAtomicWrite());
```

Comment:
> Seems like AtomicWrite would be the right thing to use. Are you just setting "no" to avoid adding a temp file when there wasn't one before?

My intent was to preserve the existing behaviour:
* No temp file
* If you have a missing header, it removes the .d file (with AtomicWrite it only removes the temp file)

But you're right that it's probably better to just use AtomicWrite and not delete a previously created .d file on failure.  There's no indication in the commit history why removing a previously existing .d file is desirable compared to just not writing the new one, and there are no test for it (I only knew about it by source inspection).  I also checked gcc on Linux, which does *not* remove an existing .d file.